core/primitives/src/trie_key.rs

```diff
@@ -50,8 +48,15 @@ pub mod col {
     /// This column id is used when storing the postponed PromiseYield receipts
     /// (`primitives::receipt::Receipt`).
     pub const PROMISE_YIELD_RECEIPT: u8 = 12;
-    /// All columns except those used for the delayed receipts queue and the yielded promises
-    /// queue, which are both global state for the shard.
+    /// Indices of outgoing receipts. A singleton per shard.
```

Comment:
Why did you decide to go with a singleton instead of having one per shard? Either works of course, I'm just curious about your reasoning.

Reply:
Yeah, both works. But storing many small values in the trie is, generally speaking, less efficient than a single larger value. It would require more IO request for RocksDB for persistence, more trie changes to track during execution, and more proofs to include in a state witness. That was my thinking.